index.js

```diff
@@ -340,7 +340,7 @@ function deleteNestedTranslation(file, translationKey, isFullPath = false) {
   //Check if this object has another translations if not delete the key.
   if (Object.keys(translationParent[attributeKey]).length === 0 || isFullPath) {
     delete translationParent[attributeKey];
-    if (objectKeys.length) {
+    if (objectKeys.length && translationParentKeys.length) {
```

Comment:
I think more precise checks would be preferable:

```suggestion
    if (objectKeys.length > 0 && translationParentKeys.length > 0) {
```